src/libexpr/eval.cc

```diff
@@ -2429,7 +2457,9 @@ bool EvalState::eqValues(Value & v1, Value & v2, const PosIdx pos, std::string_v
             return v1.string_view().compare(v2.string_view()) == 0;
 
         case nPath:
-            return strcmp(v1._path, v2._path) == 0;
+            return
+                v1._path.accessor == v2._path.accessor
```

Comment:
We might want to have a more referentially transparent equality than pointer equality at some point.
Not a problem yet while all user files are from `es.rootFS`.

Reply:
OTOH an equivalence class where "equal" files (by original path) can behave differently because one is part of a larger accessor may also be bad.
Same files in the same git tree with the same parameters would be nice to recognize as equal.
Perhaps accessors should be compared by a fingerprint? (Which may still involve a memory address when Nix functions are involved, as with future filtered input accessor?)

Reply:
Yes, on the lazy-trees branch, inputs have an optional fingerprint (e.g. the git hash) that could be used for comparing accessors. I'll add a note that we may want to use that.